lib/irc.coffee

```diff
@@ -89,6 +96,10 @@ module.exports =
       .on 'abort', @errorHandler.bind @
       .on 'join', (channel, who) =>
         console.log '%s has joined %s', who, channel if atom.config.get 'irc.debug'
+        @ircView?.addMessage 'JOINED', null, who + ' has joined ' + channel if atom.config.get('irc.showJoinMessages')
```

Comment:
@aaronholla Rather than not display the message when `showJoinMessages` is false, we should not bind the events.

IE:

```coffeescript
if atom.config.get 'irc.showJoinMessages'
  @client
    .on 'join', (channel, who) => #...
    .on 'quit', (who, reason) => #...
```

That way, if they don't have it configured, we don't have to use extra memory with event bindings that don't do anything.